modules/configuration/src/test/java/org/apache/ignite/internal/configuration/testframework/ConfigurationExtension.java

```diff
@@ -160,16 +167,31 @@ private static Object cfgValue(
 
                     change.descend(copy);
 
-                    if (superRootRef.compareAndSet(sr, copy))
-                        return completedFuture(null);
+                    if (superRootRef.compareAndSet(sr, copy)) {
+                        List<CompletableFuture<?>> futures = new ArrayList<>();
+
+                        ConfigurationNotificationsUtil.notifyListeners(
+                            sr.getRoot(rootKey),
+                            copy.getRoot(rootKey),
+                            ((DynamicConfiguration<InnerNode, ?>)cfgRef.get()),
+                            ++storageRev,
+                            futures
+                        );
+
+                        return CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
```

Comment:
In order to emulate configuration registry correctly, we should have following guaranties:
- same notification order as in real-life configuration;
- same non-fully-processing-awaiting logic of inner change notifications; 
- same fully-processing-awaiting logic of cross change notifications;
- same exception handling logic for exceptions thrown from listeners;

Reply:
Thank you!